test/test_session.py

```diff
@@ -225,16 +224,15 @@ def thread_target(op, *args):
                 thread.join()
                 self.assertIsNone(thread.exc)
             client.close()
-            lsid_set = set()
+            lsid_set.clear()
             for i in listener.results["started"]:
                 if i.command.get("lsid"):
                     lsid_set.add(i.command.get("lsid")["id"])
             if len(lsid_set) == 1:
                 succeeded = True
             else:
                 failures += 1
-        print(failures)
-        self.assertTrue(succeeded)
+        self.assertTrue(succeeded, lsid_set)
```

Comment:
This is a small improvement to include the last lsid_set in the error output when the test fails.